### PR TITLE
Add SourceKitOptions build server request

### PR DESCRIPTION
**Pull request by rmaz**
Add a request and response for SourceKitOptions, and fetch synchronously as part of the `settings` API.

### CODE REVIEW

File: Sources/BuildServerProtocol/FileOptions.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  public var uri: URL
+
+  public init(uri: URL) {
+    self.uri = uri
+  }
+}
+
+public struct SourceKitOptionsResult: ResponseType, Hashable {
+  /// The compiler options required for the requested file.
+  public var options: [String]
```
Comment (benlangmuir) [inline Sources/BuildServerProtocol/FileOptions.swift]: "compilerArguments"?

**Comment by benlangmuir** [comment]
@swift-ci please test

File: Sources/BuildServerProtocol/FileOptions.swift (hunk 1 above)
Comment (rmaz) [reply Sources/BuildServerProtocol/FileOptions.swift]: This was named after the `buildTarget/scalacOptions` request in the [BSP](https://github.com/scalacenter/bsp/blob/master/docs/bsp.md) draft, but don't feel strongly abou tit.

Comment (benlangmuir) [reply Sources/BuildServerProtocol/FileOptions.swift]: Fair enough, we can leave it.

**Pull request merged**